File: Builder/button/Button.java
package button;

abstract public class Button {
    protected int height;
    protected int width;
    protected String text;
    protected Color backgroundColor;
    protected Color textColor;
    protected String action;
    Button(ButtonBuilder builder) {
        this.height = builder.height;
        this.width = builder.width;
        this.text = builder.text;
        this.backgroundColor = builder.backgroundColor;
        this.textColor = builder.textColor;
        this.action = builder.action;
    }
    public abstract void render();
    public abstract void onClick();
}

File: BuilderKotlin/button/Button.kt
package button

abstract class Button private constructor(
    protected var height: Int? = null,
    protected var width: Int? = null,
    protected var text: String? = null,
    protected var action: String? = null,
    protected var backgroundColor: Color? = null,
    protected var textColor: Color? = null
) {
    constructor(builder: ButtonBuilder) : this(builder.height, builder.width, builder.text, builder.action, builder.backgroundColor, builder.backgroundColor)

    abstract fun render()
    abstract fun onClick()
}